onnxruntime/core/optimizer/matmul_bn_fusion.cc

```diff
@@ -0,0 +1,280 @@
+    return true;
+  }
+
+  if (!graph_utils::IsSupportedOptypeVersionAndDomain(child_node, path[0].first, path[0].second) ||
+      child_node.GetExecutionProviderType() != parent_node.GetExecutionProviderType()) {
+    return false;
+  }
+
+  /*
+  * last node in the path can have more than one output
+  * because all those outputs will be preserved by the addition of new Gemm node
+  */
+  if (path.size() > 1 && child_node.GetOutputEdgesCount() != 1) {
+    return false;
+  }
+
+  return MatchPath(child_node, path.subspan(1), *child_node.OutputNodesBegin());
+}
+
+/*
+*   Given a MatMul node, it will verify the following pattern.
+*                      MatMul
+*                        |
+*                       / \
+*                      /   \
+*                     /     \
+*               Reshape     Shape
+*                  |          |
+*             Transpose      Cast
+*                  |          |
+*        BatchNormalization  Cast
+*                  |          |
+*              Transpose      |
+*                  |         /
+*                   \       /
+*                    \     /
+*                     \   /
+*                       | 
+*                    Reshape
+* As of writing this fusion, we are being conversative in the pattern because the customer
+* model we are targeting has this exact pattern. Above pattern will evolve in the future 
+* as we tend to add separate fusion to eliminate Transpose around the BatchNormalization, 
+* update the model optimizer script to eliminate adjacent Cast operator, etc.
+* 
+* We have to match the path (MatMul->Shape->Cast->Cast->Reshape) because sub-merging the 
+* BatchNormalization into the MatMul will change MatMul's output and thus we have to make 
+* sure that MatMul's output is not used by any operator to which MatMul's output matters.
+* Other Conditions:
+*   - B tensor of MatMul should be constant.
+*   - scale, B, mean, var tensors of BatchNormalization should be constant.
+*   - Every node in the path except first and last node, should have only 1 output edge.
+*/
+bool MatmulBNFusion::SatisfyCondition(const Graph& graph, const Node& node, const logging::Logger&) const {
+  if (!graph_utils::IsSupportedOptypeVersionAndDomain(node, "MatMul", { 1, 9, 13 }) ||
+      node.GetOutputEdgesCount() != 2) {
+    return false;
+  }
+
+  auto child_node_iterator = node.OutputNodesBegin();
+  const Node& first_child_node = *child_node_iterator;
+  ++child_node_iterator;
+  const Node& second_child_node = *child_node_iterator;
+
+  std::vector<std::pair<std::string, InlinedVector<ONNX_NAMESPACE::OperatorSetVersion>>> first_path = 
+  {{"Reshape", {1, 5}},
+    {"Transpose", {1}},
+    {"BatchNormalization", {1, 6, 7}},
+    {"Transpose", {1}},
+    {"Reshape", {1, 5}}};
+
+  std::vector<std::pair<std::string, InlinedVector<ONNX_NAMESPACE::OperatorSetVersion>>> second_path =
+  {{"Shape", {1}},
+    {"Cast", {1, 6}},
+    {"Cast", {1, 6}},
+    {"Reshape", {1, 5}}};
```

Comment:
Same, opening brace should be on the same line as `=`, and closing brace should be on the next line.